### PR TITLE
DOC: Improved docs for `jnp.fft.rfftn` and `jnp.fft.irfftn`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/fft.py [hunk 1]
```diff
@@ -252,17 +252,172 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+
+  JAX implementation of :func:`numpy.fft.rfftn`.
+
+  Args:
+    a: real-valued input array.
+    s: optional sequence of integers. Specifies the effective size of the output
+      along each specified axis. If not specified, it will default to the
+      dimension of input along ``axes``.
+    axes: optional sequence of integers, default=None. Specifies the axes along
+      which the transform is computed.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:268]: We should say what default=None means in terms of which axes are transformed

File: jax/_src/numpy/fft.py [hunk 2]
```diff
@@ -252,17 +252,172 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+    Array([[[ 78.+0.j  , -12.+6.93j],
+            [ -6.+0.j  ,   0.+0.j  ]],
+    <BLANKLINE>
+           [[-36.+0.j  ,   0.+0.j  ],
+            [  0.+0.j  ,   0.+0.j  ]]], dtype=complex64)
+
+    When ``s=[3, 3, 4]``, transform is compted along ``axes (-3, -2, -1)``
+    dimension of the transform along ``axes -3 and -2`` will be ``3 and 3``,
+    along ``axis -1`` will be ``(4/2)+1) = 3`` and dimension along other axes
+    (if available) will be the same as that of input.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:304]: I'm having trouble parsing this sentence. Is there a stray copy-paste in here?

Also, I think ``` ``3 and 3`` ``` should probably be  just `3 and 3` (no need for code formatting)

File: jax/_src/numpy/fft.py [hunk 3]
```diff
@@ -252,17 +252,172 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+           [[ -7.5 -49.36j, -20.45 +9.43j,  -2.5 -16.45j],
+            [-25.5  -7.79j,  -0.6 +11.96j,  -8.5  -2.6j ],
+            [ 19.5 -12.99j,  -8.33 -6.5j ,   6.5  -4.33j]],
+    <BLANKLINE>
+           [[ -7.5 +49.36j,  12.45 -4.43j,  -2.5 +16.45j],
+            [ 19.5 +12.99j,   0.33 -6.5j ,   6.5  +4.33j],
+            [-25.5  +7.79j,   4.6  +5.04j,  -8.5  +2.6j ]]], dtype=complex64)
+
+    When ``s=[3, 5]`` and ``axes=(0, 1)``, transform is computed along
+    ``axes (0, 1)`` and size of the transform along ``axis 0`` will be ``3``,
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:321]: Remove "transform is computed along axex (0, 1)": seems somewhat repetetive

File: jax/_src/numpy/fft.py [hunk 4]
```diff
@@ -252,17 +252,172 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
           axes: Sequence[int] | None = None,
           norm: str | None = None) -> Array:
+  """Compute a multidimensional discrete Fourier transform of a real-valued array.
+
+  JAX implementation of :func:`numpy.fft.rfftn`.
+
+  Args:
+    a: real-valued input array.
+    s: optional sequence of integers. Specifies the effective size of the output
+      along each specified axis. If not specified, it will default to the
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:265]: `s` controls the effective input size, not the output size

File: jax/_src/numpy/fft.py [hunk 5]
```diff
@@ -252,17 +252,172 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+    axes: optional sequence of integers, default=None. Specifies the axes along
+      which the transform is computed.
+    norm: string, default="backward". The normalization mode. "backward", "ortho"
+      and "forward" are supported.
+
+  Returns:
+    An array containing the multidimensional discrete Fourier transform of ``a``
+    having size specified in ``s`` along the axes ``axes`` except along the axis
+    ``axes[-1]``. The size of the output along the axis ``axes[-1]`` is 
+    ``(s[-1]/2)+1``, if ``s[-1]`` is even and ``(s[-1]+1)/2``, if ``s[-1]`` is odd.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:276]: I think this is `s[-1]//2+1` in either case, yes?

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/fft.py (hunk 5 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Yes and modified accordingly.

File: jax/_src/numpy/fft.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py:265]: Modified

File: jax/_src/numpy/fft.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Removed

File: jax/_src/numpy/fft.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Modified the sentence.

File: jax/_src/numpy/fft.py [hunk 6]
```diff
@@ -252,17 +252,170 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+  Args:
+    a: input array.
+    s: optional sequence of integers. Specifies the size of the output in each
+      specified axis. If not specified, the dimension of output along axis
+      ``axes[-1]`` is ``2*(m-1)``, ``m`` is the size of input along axis ``axes[-1]``
+      and the dimension along other axes will be the same as that of input.
+    axes: optional sequence of integers, default=None. Specifies the axes along
+      which the transform is computed. If not specified, transform is computed
+      along the last ``len(s)`` axes or along all the axes, if ``s`` is also not
+      specified.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:362]:
```suggestion
      along the last ``len(s)`` axes. If neither ``axes`` nor ``s`` is specified,
      the transform is computed along all axes.
```

File: jax/_src/numpy/fft.py [hunk 7]
```diff
@@ -252,17 +252,170 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+    norm: string, default="backward". The normalization mode. "backward", "ortho"
+      and "forward" are supported.
+
+  Returns:
+    A real-valued array containing the multidimensional inverse discrete Fourier
+    transform of ``a``.
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:368]: Mention the shape here

File: jax/_src/numpy/fft.py (hunk 6 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

File: jax/_src/numpy/fft.py (hunk 7 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Updated the size of the resulting array.

**Review by jakevdp — COMMENTED** [review]
Looks good! Once you squash the commits I think we can get this merged

**Comment by rajasekharporeddy** [comment]
Thanks! Squashed into single commit.

**Review by jakevdp — APPROVED** [review]

File: jax/_src/numpy/fft.py [hunk 8]
```diff
@@ -252,17 +252,171 @@ def ifftn(a: ArrayLike, s: Shape | None = None,
...
+      which the transform is computed. If not specified, the transform is computed
+      along the last ``len(s)`` axes. If neither ``axes`` nor ``s`` is specified,
+      the transform is computed along all the axes.
+    norm: string, default="backward". The normalization mode. "backward", "ortho"
+      and "forward" are supported.
+
+  Returns:
+    An array containing the multidimensional discrete Fourier transform of ``a``
+    having size specified in ``s`` along the axes ``axes`` except along the axis
+    ``axes[-1]``. The size of the output along the axis ``axes[-1]`` is
```
Comment (jakevdp) [inline jax/_src/numpy/fft.py:277]: Looks like this line has trailing whitespace

Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Fixed and squashed

**Review by jakevdp — APPROVED** [review]

**Pull request merged**